src/HDFFile.cxx

```diff
@@ -141,141 +279,21 @@ int HDFFile::init(std::string filename,
 
   auto f1 = x;
 
+  // Traverse nexus_structure
+  // The rapidjson visitor interface is not flexible enough unfortunately
+  std::deque<SE> stack;
+
   if (nexus_structure.IsObject()) {
-    // Traverse nexus_structure
-    // The rapidjson visitor interface is not flexible enough unfortunately
-    struct SE {
-      string name;
-      Value const *jsv;
-      hid_t nxparent;
-      hid_t nxv;
-      Value::ConstMemberIterator itr;
-      bool basics;
-      string nx_type{"group"};
-      SE(string name, Value const *jsv, hid_t nxparent)
-          : name(name), jsv(jsv), nxparent(nxparent), nxv(-1),
-            itr(jsv->MemberEnd()), basics(false) {
-        if (jsv->IsObject()) {
-          itr = jsv->MemberBegin();
-        }
-      }
-    };
-    std::deque<SE> stack;
     stack.push_back(SE{"/", &nexus_structure, -1});
```

Comment:
Use `emplace_back` instead of `push_back`: `stack.emplace_back("/", &nexus_structure, -1);`